tree/ntuple/v7/inc/ROOT/RColumn.hxx

```diff
@@ -133,9 +133,22 @@ public:
 
    void AppendV(const void *from, std::size_t count)
    {
-      // TODO(jblomer): optimize
-      for (std::size_t i = 0; i < count; ++i) {
-         Append(static_cast<const unsigned char *>(from) + fElement->GetSize() * i);
+      auto src = reinterpret_cast<const unsigned char *>(from);
+      while (count > 0) {
+         std::size_t nElementsRemaining = fWritePage.GetMaxElements() - fWritePage.GetNElements();
+         if (nElementsRemaining == 0) {
+            HandleWritePageIfFull();
+            nElementsRemaining = fWritePage.GetMaxElements() - fWritePage.GetNElements();
+         }
+
+         assert(nElementsRemaining > 0);
+         auto nBatch = std::min(count, nElementsRemaining);
+
+         void *dst = fWritePage.GrowUnchecked(nBatch);
+         std::memcpy(dst, src, nBatch * fElement->GetSize());
+         src += nBatch * fElement->GetSize();
+         count -= nBatch;
+         fNElements += nBatch;
       }
```

Comment:
For me to understand: the strategy is:
1. Fill the remaining entries elements of the already allocated write page.
2. Then `nElementsRemaining == 0` and the code calls `HandleWritePageIfFull()`, which will expand the page if possible.
3. Copy elements in the expanded or flushed page.
4. Go back to 2 until done.

Wouldn't it be even better to grow the page in one go, up to the maximum unzipped page size? Then we would save on some `memcpy` while expanding. (can be a future optimization of course)

Reply:
Correct. That would be a nice optimization. I added a TODO.